aten/src/ATen/native/sparse/SparseBlas.cpp

```diff
@@ -132,8 +133,20 @@ Tensor& sparse_sampled_addmm_out_sparse_csr_cpu(
     result_sizes.push_back(self.size(-2));
     result_sizes.push_back(self.size(-1));
     at::sparse_csr::get_sparse_csr_impl(result)->resize_(self._nnz(), result_sizes);
+    result.copy_(self);
   }
-  result.copy_((self.to_dense().mul(beta).add(mat1.matmul(mat2), alpha)).sparse_mask(self));
+
+  if (mat1.numel() == 0 || mat2.numel() == 0 || result._nnz() == 0) {
+    result.mul_(beta);
+    return result;
+  }
+
+  // transpose mat2 to [b, n, k] from performance perspective.
+  // for gnn classic usage, mat2 is already stored in [b, n, k] physically,
+  // so no extra memcpy is needed.
+  auto mat2_t = mat2.transpose(-1, -2).contiguous();
+  sampled_addmm_sparse_csr_stub(kCPU, mat1.contiguous(), mat2_t, beta, alpha, result);
```

Comment:
We should probably error out on hybrid inputs.

Reply:
Could you please specify: what do you mean by "hybrid inputs" here?

Reply:
Hybrid meaning with values with dim > 1.

Reply:
The current check on `sampled_addmm` ( i did not change that) is to make sure `self` and `mat1` have the same batch size:

```C
TORCH_CHECK(
    !(self.dim() > 2 && self.sizes().slice(0, self.dim() - 2) != mat1.sizes().slice(0, mat1.dim() - 2)),
    "sampled_addmm: Expected self and mat1 to have the same batch size, but got ",
    self.sizes().slice(0, self.dim() - 2),
    " and ",
    mat1.sizes().slice(0, mat1.dim() - 2));
```

Will that be enough?

Reply:
Nope, hybrid tensors have shape `(batch_dims) + (sparse_dims) + (dense_dims)`. And the kernels you implemented assume no `dense_dims`.

Reply:
Adding
```c++
TORCH_CHECK(result.dense_dim() == 0, "sampled_addmm: Expected non-hybrid result tensor");
```
or similar to `sparse_sampled_addmm_check_inputs` ought to be sufficient to resolve this.

Reply:
Got it, thanks!